### PR TITLE
Issue #872 - Prevent user from submitting multiple applications (STAGED IN DK SANDBOX)

### LINKED ISSUE #872: Prevent users from submitting multiple applications (under certain circumstances)
### Story

As a domain applicant 
I want to be prohibited from submitting a new request when I have an request in the statuses of: _Submitted, In Review, or Action Needed_ and I have no approved requests.
so that non-verified users cannot have multiple applications.

### Acceptance Criteria

- [ ] (Condition 1) A user is _not_ permitted to **submit** a new domain request: if the user has no approved requests(s) **AND** they have a request in the _Submitted, In Review, or Action Needed_ status.
- [ ] A user in "condition 1" may always **start** a new request, which may be saved as a draft if it cannot be submitted, per these rules.
- [ ] A user in "condition 1" will see an alert, which persists on each of the pages of new requests, that informs the user that they cannot submit the request because they currently have a pending request. The pending request is identified in the alert and there will be a link in the alert to take you back to the dashboard.
- [ ] The final "Submit domain request" button on the final review page is replaced with an outlined button with the CTA 'Save and return to manage your domains.' This button takes you back to the Dashboard, and the draft is saved.
- [ ] Unit Tests are completed and up-to-date

### Issue Links
[Figma prototype with alert messaging and final button copy/styling](https://www.figma.com/proto/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?page-id=4829%3A39692&type=design&node-id=754-29647&viewport=1151%2C-64%2C0.28&t=06tdmsLsE9S8vmBx-1&scaling=min-zoom&starting-point-node-id=754%3A29647&show-proto-sidebar=1&mode=design)

Relates to: check #896

### PR DESCRIPTION
**Pull request by dave-kennedy-ecs**
## Ticket

Resolves #872 

## Changes

- Checks added for conditions that prevent user from submitting multiple applications
- Informational message displayed when conditions are met
- Submit button is changed on review page in application when conditions are met

### Alerts refactoring

Full size alert is only used for informational purposes and never for alerts that are triggered by form interaction. 
Alerts that are triggered by form interaction are slim. See [slack thread](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1701257550133219) for reference.

- There are large triggered alerts. The one we're using on 872 is one such alert. We can refactor those instances to use a large variant (for future documentation, this alert will be `pending_requests_message` in the code)
- On domain management pages: There are currently 21 alerts triggered by form success/error/warning. They are all small one liners except for this guy (see screenshot)
- Revised persistent alerts on DNSSEC and nameservers to use full variant (screenshots)
- Kept DS Data persistent alert slim

## Context for reviewers

**This PR is staged in [DK Sandbox](https://getgov-dk.app.cloud.gov/)**

Here is the [FIGMA link](https://www.figma.com/proto/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?page-id=4829%3A39692&type=design&node-id=754-30047&viewport=1151%2C-64%2C0.28&t=06tdmsLsE9S8vmBx-1&scaling=min-zoom&starting-point-node-id=754%3A29647&show-proto-sidebar=1) which shows the desired user interactions.

**Design review:**
The first test assumes that you have at least one domain in your account.
For the first test, click 'Start a new domain request' on the home page.  When you do so, you should see no blue info message at the top of the page.  If no info message, then this test is successful, and go through the setup outlined in the SetUp section below for other tests.

Once you follow the setup below, you should have zero domains on your home page.  Under Domains, it should read 'You don't have any registered domains yet'.  And you should have at least one domain requests with one of the following statuses: IN_REVIEW, APPROVED, ACTION_NEEDED

For the second test, you will be testing that the presence of zero approved domains, and one or more requests in the above states, will produce an blue info message at the top of the application screen.
Go to the home page, and click 'Start a new domain request'. Test that you receive an info message 'You cannot submit this request yet' at the top of the application page.  Continue to fill out the application, and verify that this message appears at the top of every page in the application process.  On the review page, the button should be an outlined button that reads, 'Save and return to manage your domains'.  When this is clicked, changes will be saved and you will be directed to home page.  You should now see the application you created, and the status should be Started.  If you click Edit, you should see your application and the same info message at the top of the page.

For the third test, test that you don't receive the info message when you click on a domain application with a Status of Action Needed.  On the home page, find a domain request with Status of 'Action Needed'.  When you click 'Edit' you should see the domain application page, and there should be no blue info message at the top of the screen.

**Developer Code Review:**
templates/application_form.html - added a new conditional button when user should be blocked from submitting; button has value of save_and_return, which is handled in the view
views/application.py - added methods which determine if approved domain applications exist, if domains exist, and if domain applications with status in (IN_REVIEW, ACTION_NEEDED, or SUBMITTED) exist; info message added in get() if conditions met; variable added to context in order to display the proper submit button; post set up to handle new submit button

## Setup

It is likely that you already have at least one domain in your account.  If so, perform the first test above before following the rest of the setup.

The setup for this can be quite involved, depending upon how many domains and domain applications you have associated with your user.

For each domain which is showing on your home page, you need to perform the following in django admin:
- delete each contact associated with the domain
- delete the domain
- delete the domain application associated with the domain

Once you have deleted all domains associated with the user, you need to make sure there exists at least one domain application that is either IN_REVIEW, APPROVED, or ACTION_NEEDED.  In order to test ACTION_NEEDED, you should have at least one domain application with ACTION_NEEDED status.

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [x] Pulled this branch locally and tested it
- [x] Reviewed this code and left comments
- [x] Checked that all code is adequately covered by tests
- [x] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [x] All new functions and methods are commented using plain language
- [x] Interactions with external systems are wrapped in try/except
- [x] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/views/application.py [hunk 1]
```diff
@@ -266,6 +285,38 @@ def get_forms(self, step=None, use_post=False, use_db=False, files=None):
...
+
+    def pending_applications(self):
+        """Returns a List of user's applications with one of the following states:
+        SUBMITTED, IN_REVIEW, ACTION_NEEDED"""
+        # if the current application has ACTION_NEEDED status, this check should not be performed
+        if self.application.status == DomainApplication.ACTION_NEEDED:
+            return []
+        check_statuses = [DomainApplication.SUBMITTED, DomainApplication.IN_REVIEW, DomainApplication.ACTION_NEEDED]
+        filter_conditions = Q(creator=self.request.user) & Q(status__in=check_statuses)
+        return DomainApplication.objects.filter(filter_conditions)
```
Comment (zandercymatics) [inline src/registrar/views/application.py:318]:
```suggestion
    return DomainApplication.objects.filter(
        creator=self.request.user,
        status__in=check_statuses
    )
```
While this works, I think using Q might be a bit overkill here. The filter above should be syntactically equivalent (though I wouldn't commit that directly, the spacing is broke on the github end).

I would recommend this just for legibility purposes

**Review by zandercymatics — COMMENTED** [review]
Code looks good, just had one minor nitpick. Will test tomorrow once your sandbox is up

File: src/registrar/tests/test_views.py [hunk 2]
```diff
@@ -144,6 +144,42 @@ def test_application_form_empty_submit(self):
         result = page.form.submit()
         self.assertIn("What kind of U.S.-based government organization do you represent?", result)
 
+    def test_application_multiple_applications_exist(self):
+        """Test that an info message appears when user has multiple applications already"""
+        # create and submit an application
+        contact = Contact.objects.create()
```
Comment (neilmb) [inline src/registrar/tests/test_views.py:150]: **question (non-blocking)**: There is a `create_full_dummy_domain_application` that might streamline this first section of the test?

**Review by neilmb — COMMENTED** [review]
It's a theme, just a nit pick in the unit testing, otherwise LGTM.

File: src/registrar/views/application.py [hunk 3]
```diff
@@ -218,6 +219,22 @@ def get(self, request, *args, **kwargs):
...
         context["forms"] = self.get_forms()
+
+        # if pending requests exist and user does not have approved domains,
+        #     present message that domain application cannot be submitted
+        pending_requests = self.pending_requests()
+        if len(pending_requests) > 0:
+            message_header = "You cannot submit this request yet"
+            message_content = (
+                f"<h4 class='usa-alert__heading'>{message_header}</h4>"
+                "<p>New domain requests cannot be submitted until we have finished reviewing your pending request: "
```
Comment (rachidatecs) [inline src/registrar/views/application.py:230]:
```suggestion
                "<p class="margin-bottom-0">New domain requests cannot be submitted until we have finished reviewing your pending request: "
```

**Review by rachidatecs — COMMENTED** [review]
For whoever wants to review on local, suggest changing line 178 in Domain model to `return True`

File: src/registrar/tests/test_views.py (hunk 2 above)
Comment (dave-kennedy-ecs) [reply src/registrar/tests/test_views.py]: thanks @neilmb I streamlined the first section of the test at your suggestion

**Review by zandercymatics — APPROVED** [review]
LGTM. Works well

**Comment by Katherine-Osos** [comment]
@dave-kennedy-ecs The functionality all works as expected, but the margin/padding for the info message in incorrect. Looks like you're using the "slim" version of the USWDS component rather than the "full" version. Can you update this to use the "full" version? 

Incorrect padding and margin

**Review by gabydisarli — APPROVED** [review]

**Pull request merged**